src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -356,11 +356,18 @@ public BitbucketCommit call() throws Exception {
     }
 
     private void setupClosureForPRBranch(BitbucketServerPullRequest pr) {
-        BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
-
-        branch = (BitbucketServerBranch) pr.getDestination().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+        try {
+            BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+            branch = (BitbucketServerBranch) pr.getDestination().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+        } catch (Exception e) {
```

Comment:
Do you really want to catch ALL exceptions here?

Reply:
Yup, I missed that 😱

Reply:
@2improveIT ping

Reply:
adapted the code